lib/Models/ItemSearchProviders/NumericIndex.ts

```diff
@@ -63,18 +64,17 @@ export default class NumericIndex implements IndexBase<NumericSearchQuery> {
     const idValuePairs = await this.idValuePairs;
     const startValue = value.start === undefined ? range.min : value.start;
     const endValue = value.end === undefined ? range.max : value.end;
-    const i = binarySearch(
+    const startIndex = sortedIndexBy(
       idValuePairs,
-      { value: startValue },
-      (a, b) => a.value - b.value
+      { dataRowId: 0, value: startValue },
+      entry => entry.value
     );
-    const j = binarySearch(
-      idValuePairs,
-      { value: endValue },
-      (a, b) => a.value - b.value
-    );
-    const startIndex = i < 0 ? ~i : i;
-    const endIndex = j < 0 ? ~j : j;
+    const endIndex =
+      sortedLastIndexBy(
+        idValuePairs,
+        { dataRowId: 0, value: endValue },
+        entry => entry.value
+      ) - 1;
```

Comment:
This seems to work well if the value is present and if the value isn't present, but I'd like to have more tests on this.